src/k8s/pkg/k8sd/app/hooks_node_ready.go

```diff
@@ -0,0 +1,42 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/canonical/k8s/pkg/log"
+	"github.com/canonical/k8s/pkg/utils"
+	"github.com/canonical/microcluster/v2/state"
+)
+
+// onNodeReady is called when the node is ready, right after the wait group is released.
+// The node is ready if:
+// - the microcluster database is accessible
+// - the kubernetes endpoint is reachable.
+// Note that this is not a microcluster hook, but a custom k8sd hook.
+func (a *App) onNodeReady(ctx context.Context, s state.State) error {
```

Comment:
```suggestion
func (a *App) onK8sdNodeReady(ctx context.Context, s state.State) error {
```
It might be handy to clarify it is a k8sd hook in the function name.